refactor: dto 네이밍 수정 및 TeamQueryService 리팩터링

OzRagwort opened the pull request:
## 구현 기능
- JDBC로 조회 후 응답용 DTO는 `XXXQueryResult` 로 통일 (dto.query 패키지에 위치시킨다.)
- 일반적인 요청 dto → XXXRequest
- 일반적인 응답 dto → XXXResponse
- 요청, 응답 dto는 분리한다.

## 논의하고 싶은 내용
~- 목록 조회에 사용되는 DTO의 네이밍 규칙을 `XXXResponses`에서 `XXXListResponses`처럼 통일 시켜봄 - 릭 피드백 반영~
~  - 피드백 목록조회 dto가 `FeedbackResponses`에서 `FeedbackListResponses`처럼 된다는 의미~
~  - 기존 팀 목록 조회도 `TeamListDto`처럼 사용중이었음~
- 팀 목록 조회 DTO가 `TeamListDto` -> `TeamListResponse`로 변경
- `TeamQueryResults`가 `List<TeamListDetailQueryResult>`을 가지도록 함
- `InterviewQuestionSearchResultDto`와 `InterviewQuestionSearchResultsDto`가 `InterviewQuestionSearchResponse`와 `InterviewQuestionSearchResponses`로 수정
- 그 외엔 거의 대부분 단순 네이밍 변경(dto -> XXXRequest, XXXResponse, XXXQueryResult) 입니다
- 정적 팩토리 메서드 대부분 제거
---

## 팀과 관련된 빙빙도는 네이밍 정리 😵‍💫

- TeamResponse(팀 엔티티의 필드만 가지는 DTO)
- TeamListResponse(팀 목록 조회)
- TeamListResponses(팀 목록 조회 시 하나의 단위)
- TeamDetailResponse(팀 상세 조회)
- TeamDetailQueryResult(팀 상세 조회 쿼리 결과)
- TeamDetailQueryResults(팀 상세 조회 쿼리 결과 모음)
- TeamListQueryResult(팀 목록 조회 쿼리 결과)
- TeamListQueryResults(팀 목록 조회 쿼리 결과 모음)

---
- Participant의 team과 member의 연관관계만 없애면 SimpleParticipant가 됨
  - Participants와 SimpleParticipants도 비슷
- TeamStatus.of()가 Team.status()와 똑같음
  - 중복 제거 필요

wyRagwort committed:
refactor: dto 이름 수정

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
Merge branch 'develop' into refactor/all-dto-naming

Signed-off-by: ozragwort <ozragwort@gmail.com>

# Conflicts:
#	backend/src/main/java/com/woowacourse/levellog/team/application/TeamQueryService.java
#	backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllParticipantQueryResult.java

wyRagwort committed:
refactor: 팀관련 DTO 수정

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
refactor: 변수 네이밍 수정

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
style: 컨벤션 적용

Signed-off-by: ozragwort <ozragwort@gmail.com>

nailseong commented on backend/src/main/java/com/woowacourse/levellog/feedback/dto/response/FeedbackResponses.java:
```diff
@@ -11,7 +11,7 @@
 @AllArgsConstructor
 @NoArgsConstructor(access = AccessLevel.PRIVATE)
-public class FeedbacksDto {
+public class FeedbackResponses {
```
`XXXResponses` -> `XXXListResponses`는 어떨까요?
XXXResponses와 XXXResponse가 구분이 잘 안되더라구요.

nailseong commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllParticipantQueryResult.java:
```diff
@@ -1,11 +1,12 @@
 
 @Getter
-public class AllParticipantDto {
+public class AllParticipantQueryResult {
```
1. Participant보다는 Team이라는 키워드를 넣기
2. 해당  DTO를 생성하는 쿼리의 목적이 묻어나는 이름 (e.g. 팀 상세 조회, 팀 목록 조회 등)

nailseong commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllSimpleParticipantQueryResult.java:
```diff
@@ -1,12 +1,12 @@
 @Getter
 @AllArgsConstructor
-public class AllSimpleParticipantDto {
+public class AllSimpleParticipantQueryResult {
```
`All`, `Simple`과 같은 키워드는 의미를 잘 나타내지 못한다고 생각해요.

nailseong commented on backend/src/main/java/com/woowacourse/levellog/team/dto/response/TeamListResponses.java:
```diff
@@ -0,0 +1,13 @@
+
+@Getter
+@AllArgsConstructor
+public class TeamListResponses {
```
TeamListResponse가 더 좋아보여욤

nailseong reviewed (commented):
고생했습니다 알린!!
Team쪽 dto이름은 팀원들과 논의를 해보면 좋을것 같아요.

wyRagwort committed:
fix: API 에 맞도록 팀 상세 조회 시 멤버 프로필 url 추가

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
refactor: 변수명, 메서드 명 수정

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
refactor: 목록 조회 네이밍을 XXXResponses에서 XXXListResponses로 통일

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
refactor: queryResult 관련 코드 리팩터링

- 일급 컬렉션 생성
- service 내 로직 QueryResult로 이동
- 네이밍 수정
  - 목록 조회 관련 : XXXListResponse
  - 상세 조회 관련 : XXXDetailResponse

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
fix: TimeStandard를 사용하여 시간을 구하도록 수정

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
refactor: 팀 목록 조회 관련 dto 수정

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
refactor: 컨벤션 적용

Signed-off-by: ozragwort <ozragwort@gmail.com>

SuyeonChoi commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResults.java:
```diff
@@ -0,0 +1,69 @@
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AllTeamDetailQueryResults {
```
목록을 반환하는 query DTO의 클래스들도 `s`가 아닌 `AllTeamDetailListQueryResult`로 구분하는건 어떨까요?

OzRagwort replied:
전체적으로 싹 바꿨습니다!

OzRagwort replied:
`s`를 추가하도록 수정했습니다

SuyeonChoi reviewed (approved):
알린 DTO 네이밍 변경하느라 수고하셨습니다👏👏
네이밍에 대해서 전 크게 신경쓰이는 부분은 없는거 같아요~! 대부분 무난무난하게 이해됐습니다👍
논의하고 싶은 내용에 대해 제 생각을 남기고 갑니당,,
제가 혹시라도 잘못이해한 부분이 있다면 말해주세요! 바로 디엠 주셔도 됩니다,,

그리구 저도 나름 빠르게 리뷰한거 같은데.... 햄버거.. 줄서봅니다^.^🍔키키
우리에게 머지데이(D-5)도 있으니까,,,다른 리뷰어들도 리뷰해줄때까지 조금만 더 기다려보죵,,
그럼 화이팅화이팅화이팅!!!🔊🔊🔊

---
알린이 논의하고 싶다는 내용에 대한 제 생각입니다😊😊
> 목록 조회에 사용되는 DTO의 네이밍 규칙을 XXXResponses에서 XXXListResponses처럼 통일 시켜봄 - 릭 피드백 반영
피드백 목록조회 dto가 FeedbackResponses에서 FeedbackListResponses처럼 된다는 의미
기존 팀 목록 조회도 TeamListDto처럼 사용중이었음
팀 목록 조회 DTO가 TeamListDto -> TeamListResponses로 변경되었는데 네이밍 어떠신가요?
InterviewQuestionSearchResultDto와 InterviewQuestionSearchResultsDto가 InterviewQuestionSearchResponse와 InterviewQuestionSearchResponses로 수정 되었는데 네이밍 어떠신가요?

혹시 `XXXListResponse`는 어떨까요? List에서 복수를 표현해줬는데 Response에서 또다시 `s`로 나타내는게 불필요하다고 느껴지네요,,! 기존에도 `TeamListDtos`가 아닌 `TeamListDto`를 사용했던 것처럼요!

2yujeong commented on backend/src/main/java/com/woowacourse/levellog/interviewquestion/domain/InterviewQuestionQueryRepository.java:
```diff
@@ -22,9 +22,9 @@ public InterviewQuestionQueryRepository(final JdbcTemplate jdbcTemplate) {
-    public List<InterviewQuestionSearchResultDto> searchByKeyword(final String keyword, final Long memberId,
-                                                                  final Long size, final Long page,
-                                                                  final InterviewQuestionSort sort) {
+    public List<InterviewQuestionSearchResponse> searchByKeyword(final String keyword, final Long memberId,
```
JDBC 조회 결과로 바로 응답하기 위한 DTO니까 이것도 `XXXQueryResult`여야 하지 않을까요?

OzRagwort replied:
그렇게 수정했습니다. !

2yujeong reviewed (changes requested):
고생하셨습니다 알린👍🏼
> - `InterviewQuestionSearchResultDto`와 `InterviewQuestionSearchResultsDto`가 `InterviewQuestionSearchResponse`와 `InterviewQuestionSearchResponses`로 수정 되었는데 네이밍 어떠신가요?

이 부분은 PR 코멘트 내용 참고해서 RC 하나 남겼는데 확인해주세용

\+ 저도 페퍼 의견처럼 `XXXListResponse`가 좋을 것 같습니다!

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResult.java:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    public TeamResponse getTeamResponse() {
+        return TeamResponse.from(id, title, place, startAt, teamProfileUrl, interviewerNumber, isClosed);
```
AllTeamDetailQueryResult에서 다양한 DTO로 변경되는군요!

한가지 의문은 `TeamResponse` 로 전환하는만 정적팩토리를 쓰네요. 생성자와 정적 팩토리를 결정하는 알린만의 기준이 있을까요?

OzRagwort replied:
기존 코드가 그렇게 되어있어서 사용했습니다.

생성자를 사용하는 경우
- 기본 생성자
- 필수적인 필드가 모두 파라미터로 들어간 생성자

정적 팩토리 메서드를 사용하는 경우
- 다른 도메인이 파라미터로 오는 경우
- 그 외 대부분의 경우

하지만 지금은 기존에 `TeamResponse`가 `@AllArgsConstructor(access = AccessLevel.PRIVATE)`로 생성자가 private이었습니다. 

따라서 정적 팩토리 메서드를 사용하였습니다.

OzRagwort replied:
젤 위에 적어주신 `다른 리뷰어분들이 TeamListResponses -> TeamListResponse 로 바꾸자는 의견 달아주셨는데 기존 TeamListResponse 는 뭘로 바꿔야할까요?? 🥹`는 

팀목록조회
`TeamListResponse`가 `List<TeamListDetailResponse>`를 가지는 식으로 수정했습니다.

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResults.java:
```diff
@@ -0,0 +1,69 @@
+
+    private TeamStatus toTeamStatus(final AllTeamDetailQueryResult result, final LocalDateTime nowTime) {
+        return result.getAllTeamListQueryResult()
+                .getTeamStatus(nowTime);
```
getAllTeamListQueryResult로 한번 전환하는 것보다는 TeamStatus를 바로 얻어내는 것이 디미터 법칙도 만족하고 객체에 메세지를 잘 전달하는 것 같아보이네요.
현재 맞춤 쿼리를 제작하고 있으니 `AllTeamDetailQueryResult` -> `AllTeamListQueryResult`로 갈 경우도 많이 없을 것 같으니 `getAllTeamListQueryResult`가 필요해보이진 않네요. 🧚‍♀️

OzRagwort replied:
맞아요 꼭 필요하지 않는 부분입니다

필드가 너무 많고 다른 dto와 중복으로 사용되는 것들이 너무 많아보여서 저렇게 수정해봤는데 다시 잘 리팩해봐야겠네요

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResults.java:
```diff
@@ -0,0 +1,69 @@
+        return results.isEmpty();
+    }
+
+    public TeamDetailResponse toResponses(final Long memberId, final LocalDateTime time) {
```
toResponse
```suggestion
    public TeamDetailResponse toResponse(final Long memberId, final LocalDateTime time) {
```
TeamDetailResponse 타입으로 변하니 위 네이밍이 적절해보이네요. 😊

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/application/TeamQueryService.java:
```diff
@@ -34,81 +25,35 @@ public class TeamQueryService {
+                condition.isClosed(), size, page * size);
 
-        return new TeamListDto(teamSimpleDtos);
+        return TeamListResponses.from(results, timeStandard.now());
```
TeamQueryService에서 `findAll`,`findAllByMemberId`는 정적 팩토리 메서드를 활용하여 Responses를 만들고 있고 `findByTeamIdAndMemberId` 에서는 일급컬렉션에 메세지를 보내서 만들어내고 있네요.

알린이 설정한 기준이 무엇인지 궁금하고 어떤 쪽으로 맞춰야 좋을지 한번 고민해보면 좋을 것 같아요 😊

OzRagwort replied:
일급 컬렉션에서 메시지를 보내는 방식으로 하나의 방식으로 통일해보겠습니다. 

---

## 고민했던 것

제 기준은

1. 코드가 깔끔해지고 역할이나 책임의 분리가 되는 방식. 큰 상관 없다면 기존의 있는 방식
2. 일급 컬렉션에서 결과를 만들어 반환하는 방식
3. 정적 팩토리 메서드를 이용하는 방식
인 것 같습니다. 

Service에 로직을 이동시키면서 리팩하는 방식은 아래 2가지 방식이 있을 것 같습니다. 
1. TeamListResponse에서 정적 팩터리 메서드를 사용하자. 
2. List<AllTeamListQueryResult>를 묶는 일급 컬렉션을 만들어 toTeamListResponse() 메서드를 만들자.

팀 목록 조회에서는 1번, 팀 상세 조회에서는 2번 방식을 이용한 것 같아서 2번 방식으로 통일하도록 수정하겠습니다.

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResults.java:
```diff
@@ -0,0 +1,69 @@
+
+public class AllTeamDetailQueryResults {
+
+    List<AllTeamDetailQueryResult> results;
```
```suggestion
    private final List<AllTeamDetailQueryResult> results;
```
👀

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResults.java:
```diff
@@ -0,0 +1,69 @@
+        return filteredParticipants.stream()
+                .filter(it -> !it.getSimpleParticipant().isWatcher())
+                .map(AllTeamDetailQueryResult::getParticipantResponse)
+                .collect(Collectors.toList());
```
```suggestion
        return filteredParticipants.stream()
                .filter(it -> it.getSimpleParticipant().isParticipant())
                .map(AllTeamDetailQueryResult::getParticipantResponse)
                .collect(Collectors.toList());
```
😊
개인적으로는 SimpleParticipant로 한번 거쳐가지 않아도 될 것 같네요.

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResults.java:
```diff
@@ -0,0 +1,69 @@
+        final List<SimpleParticipant> participants = filteredParticipants.stream()
+                .map(AllTeamDetailQueryResult::getSimpleParticipant)
+                .collect(Collectors.toList());
+        return new SimpleParticipants(participants);
```
```suggestion
        return filteredParticipants.stream()
                .map(AllTeamDetailQueryResult::getSimpleParticipant)
                .collect(Collectors.collectingAndThen(Collectors.toList(), SimpleParticipants::new));
```
이것도 가능하겠군요 😊

OzRagwort replied:
collectingAndThen 감사합니다 배웠네유

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResults.java:
```diff
@@ -0,0 +1,69 @@
+        );
+    }
+
+    private SimpleParticipants toSimpleParticipants(final List<AllTeamDetailQueryResult> filteredParticipants) {
```
개인적으로는 filteredParticipants라는 말이 납득이 잘 안되네요. results를 그냥 전달해주는건데요~ 
그리고 필드로 존재하는데 꼭 인자로 받아야하나요?👀
이 외에 다른 메서드에서도 필드값을 주입해서 사용하고 있네요😢

OzRagwort replied:
기존의 코드를 그대로 옮기면서 생각하지 못했네요 찾아주셔서 감사합니다

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResult.java:
```diff
@@ -0,0 +1,66 @@
+        this.isWatcher = isWatcher;
+    }
+
+    public AllTeamListQueryResult getAllTeamListQueryResult() {
```
여기 있는 메서드 모두 protected로 해도 될거같은데요 👀

OzRagwort replied:
그렇군요 다 같은 패키지 내에서 사용되고 있어서 protected로 수정해주었습니다.

kbsat commented on backend/src/test/java/com/woowacourse/levellog/domain/TeamQueryRepositoryTest.java:
```diff
@@ -36,11 +36,17 @@ void findAllParticipants() {
 
         // when
-        final List<AllParticipantDto> actual = teamQueryRepository.findAllByTeamId(team.getId(), roma.getId());
+        final List<AllTeamDetailQueryResult> actual = teamQueryRepository.findAllByTeamId(team.getId(), roma.getId());
```
이 테스트 어떻게 돌아가죠?
그리고 해당 테스트 메서드의 네임과 DislayName이 잘못 설정된 것 같아요.

OzRagwort replied:
해당 dto가 필드를 DTO로 가지도록 수정하면서 이렇게 수정했는데 다시 바꾸면서 then 절을 다시 수정했습니다.

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailQueryResult.java:
```diff
@@ -0,0 +1,66 @@
+    private final boolean isHost;
+    private final boolean isWatcher;
+
+    public AllTeamDetailQueryResult(final Long id, final String title, final String place, final LocalDateTime startAt,
```
builder 한번 써보시겠어요? 😉

OzRagwort replied:
개인적으로 lombok의 builder가 편하긴 한데 선호하지 않는 편이기는 해요 롬복에 의존된 코드가 되는 것 같아서요 ㅠㅠ.. 

전체적으로 이런 경우 다 빌더를 사용한다면 여기에서도 사용해도 되겠지만 아니라면 다같이 사용하지 않는건 어떨까요?

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/response/TeamListResponses.java:
```diff
@@ -0,0 +1,31 @@
+    private List<TeamListResponse> teams;
+
+    public static TeamListResponses from(final List<AllTeamListQueryResult> allParticipants, final LocalDateTime time) {
+        final List<TeamListResponse> responses = allParticipants.stream()
+                .collect(Collectors.groupingBy(
+                        AllTeamListQueryResult::getId,
+                        LinkedHashMap::new,
+                        Collectors.toList()))
+                .values()
+                .stream()
+                .map(it -> TeamListResponse.of(it, it.get(0).getTeamStatus(time)))
+                .collect(Collectors.toList());
+
+        return new TeamListResponses(responses);
```
위에처럼 collectingAndThen 사용해보십숑 😊

kbsat reviewed (changes requested):
![image](https://user-images.githubusercontent.com/52696169/194719538-b8cab510-febd-4628-9b2f-7487ef1419d5.png)

안녕하세요 알린. 리뷰어 로마입니다. 🧚‍♀️
고생해주셨네요 덕분에 QueryService가 깔끔해졌어요! 
로직 범벅이었는데 다 적절하게 객체 안으로 넣어주었기 때문이겠죠? ㅋㅋ 문득 위 짤이 생각나서 가져와봤습니다
Team 도메인 내에 service와 queryResult 위주로 리뷰 달아보았으니 보시고 반영해주세요~😊

### 논의 사항

> 목록 조회에 사용되는 DTO의 네이밍 규칙을 XXXResponses에서 XXXListResponses처럼 통일 시켜봄 
> 팀 목록 조회 DTO가 TeamListDto -> TeamListResponses로 변경되었는데 네이밍 어떠신가요?
- 저는 좋은 것 같은데욤 
- 다른 리뷰어분들이 `TeamListResponses` -> `TeamListResponse` 로 바꾸자는 의견 달아주셨는데 기존 `TeamListResponse` 는 뭘로 바꿔야할까요?? 🥹

> InterviewQuestionSearchResultDto와 InterviewQuestionSearchResultsDto가 InterviewQuestionSearchResponse와 InterviewQuestionSearchResponses로 수정 되었는데 네이밍 어떠신가요?
- 이건 이브의 의견과 동일합니다.

wyRagwort committed:
refactor: 리뷰 반영

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
Merge remote-tracking branch 'origin/develop' into refactor/all-dto-naming

# Conflicts:
#	backend/src/main/java/com/woowacourse/levellog/authentication/application/OAuthService.java
#	backend/src/main/java/com/woowacourse/levellog/team/dto/AllParticipantDto.java
#	backend/src/test/java/com/woowacourse/levellog/acceptance/OAuthAcceptanceTest.java

wyRagwort committed:
Merge remote-tracking branch 'origin/develop' into refactor/all-dto-naming

# Conflicts:
#	backend/src/main/java/com/woowacourse/levellog/feedback/application/FeedbackService.java
#	backend/src/main/java/com/woowacourse/levellog/feedback/dto/FeedbackDto.java
#	backend/src/main/java/com/woowacourse/levellog/feedback/dto/request/FeedbackContentRequest.java
#	backend/src/main/java/com/woowacourse/levellog/feedback/presentation/FeedbackController.java
#	backend/src/main/java/com/woowacourse/levellog/interviewquestion/application/InterviewQuestionService.java
#	backend/src/main/java/com/woowacourse/levellog/interviewquestion/domain/InterviewQuestionQueryRepository.java
#	backend/src/main/java/com/woowacourse/levellog/interviewquestion/dto/InterviewQuestionDto.java
#	backend/src/main/java/com/woowacourse/levellog/interviewquestion/dto/response/InterviewQuestionListResponse.java
#	backend/src/main/java/com/woowacourse/levellog/interviewquestion/presentation/InterviewQuestionController.java
#	backend/src/main/java/com/woowacourse/levellog/interviewquestion/presentation/InterviewQuestionSearchController.java
#	backend/src/main/java/com/woowacourse/levellog/levellog/application/LevellogService.java
#	backend/src/main/java/com/woowacourse/levellog/levellog/dto/response/LevellogWithIdResponse.java
#	backend/src/main/java/com/woowacourse/levellog/levellog/presentation/LevellogController.java
#	backend/src/main/java/com/woowacourse/levellog/member/application/MemberService.java
#	backend/src/main/java/com/woowacourse/levellog/member/presentation/MyInfoController.java
#	backend/src/main/java/com/woowacourse/levellog/prequestion/application/PreQuestionService.java
#	backend/src/main/java/com/woowacourse/levellog/prequestion/presentation/PreQuestionController.java
#	backend/src/main/java/com/woowacourse/levellog/team/application/TeamQueryService.java
#	backend/src/main/java/com/woowacourse/levellog/team/application/TeamService.java
#	backend/src/main/java/com/woowacourse/levellog/team/domain/TeamQueryRepository.java
#	backend/src/main/java/com/woowacourse/levellog/team/presentation/TeamController.java
#	backend/src/test/java/com/woowacourse/levellog/application/FeedbackServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/application/InterviewQuestionServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/application/LevellogServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/application/MemberServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/application/OAuthServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/application/PreQuestionServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/application/ServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/application/TeamQueryServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/application/TeamServiceTest.java
#	backend/src/test/java/com/woowacourse/levellog/domain/FeedbackTest.java
#	backend/src/test/java/com/woowacourse/levellog/domain/InterviewQuestionLikesRepositoryTest.java
#	backend/src/test/java/com/woowacourse/levellog/domain/RepositoryTest.java
#	backend/src/test/java/com/woowacourse/levellog/domain/TeamQueryRepositoryTest.java
#	backend/src/test/java/com/woowacourse/levellog/presentation/LevellogControllerTest.java

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamDetailListQueryResult.java:
```diff
@@ -11,24 +10,24 @@
         final AllTeamDetailQueryResult result = results.get(0);
 
         final TeamResponse teamResponse = result.getTeamResponse();
         final TeamStatus status = toTeamStatus(result, time);
```
29번째 라인 : 
toTeamStatus 보다 result.getTeamStatus(time); 이 더 명료한 것 같아요

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/query/AllTeamListQueryResult.java:
```diff
@@ -1,24 +1,30 @@
+        this.results = results;
+    }
+
+
```
한칸만 뛰십쇼 🤔

kbsat commented on backend/src/main/java/com/woowacourse/levellog/team/dto/AllParticipantDto.java:
```diff
@@ -1,53 +0,0 @@
-package com.woowacourse.levellog.team.dto;
```
이것도 지워주세요~! 😊

kbsat commented on backend/src/main/java/com/woowacourse/levellog/prequestion/dto/request/PreQuestionWriteRequest.java:
```diff
@@ -14,13 +14,13 @@
 
-    public static PreQuestionWriteDto from(final String content) {
-        return new PreQuestionWriteDto(content);
+    public static PreQuestionWriteRequest from(final String content) {
+        return new PreQuestionWriteRequest(content);
```
정적 팩토리 메서드를 사용하는 이유가 없는 것 같아요 ㅜ.ㅜ

OzRagwort replied:
제말이요...

wyRagwort committed:
fix: 제대로 되지 않은 머지 파일 수정

Signed-off-by: ozragwort <ozragwort@gmail.com>

wyRagwort committed:
refactor: dto를 List로 가지는 경우 XXXs로 명명하도록 수정

- TeamListResponse -> TeamResponses

Signed-off-by: ozragwort <ozragwort@gmail.com>

kbsat reviewed (changes requested):
안녕하세요 알린, 리뷰어 로마입니다.
잘 고쳐주셨네요. 소소한 부분 리뷰해보았으니 반영부탁드려요 😊

wyRagwort committed:
refactor: 불필요한 정적 팩토리 메서드 제거

Signed-off-by: ozragwort <ozragwort@gmail.com>

2yujeong reviewed (approved):
👍🏼

kbsat reviewed (approved):
알린 고생하셨습니다 🙉🥹🙉 
이만 머지하겠습니다.
